hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/NextKSketch.java

```diff
@@ -0,0 +1,154 @@
+     */
+    private ObjectArrayColumn mergeColumns(@NonNull final IColumn left,
+                                           @NonNull final IColumn right,
+                                           @NonNull final List<Integer> mergeOrder) {
+        final int size = Math.min(this.maxSize, mergeOrder.size());
+        final ObjectArrayColumn merged = new ObjectArrayColumn(left.getDescription(), size);
+        int i = 0, j = 0, k = 0;
+        while (k < size) {
+            if (mergeOrder.get(k) == -1) {
+                merged.set(k, left.getObject(i));
+                i++;
+            } else if (mergeOrder.get(k) == 1) {
+                merged.set(k, right.getObject(j));
+                j++;
+            } else {
+                merged.set(k, right.getObject(j));
+                i++;
+                j++;
+            }
+            k++;
+        }
+        return merged;
+    }
+
+    /**
+     * Given two Columns containing counts left and right, merge them to a single Column, using an
+     * Integer array mergeLeft which represents the order in which elements merge.
+     * @param left       The left counts.
+     * @param right      The right counts.
+     * @param mergeOrder The order in which to merge the two columns.
+     * @return The merged counts.
+     */
+    private List<Integer> mergeCounts(@NonNull final List<Integer> left,
+                                      @NonNull final List<Integer> right,
+                                      @NonNull final List<Integer> mergeOrder) {
+        final int size = Math.min(this.maxSize, mergeOrder.size());
+        final List<Integer> mergedCounts = new ArrayList<Integer>(mergeOrder.size());
+        int i = 0, j = 0, k = 0;
+        while (k < size) {
+            if (mergeOrder.get(k) == -1) {
+                mergedCounts.add(left.get(i));
+                i++;
+            } else if (mergeOrder.get(k) == 1) {
+                mergedCounts.add(right.get(i));
+                j++;
+            } else {
+                mergedCounts.add(left.get(i) + right.get(j));
+                i++;
+                j++;
+            }
+            k++;
+        }
+        return mergedCounts;
+    }
+
+    /**
+     * Add two NextK Lists, merging counts of identical rows.
+     * @param left The left TopK list
+     * @param right The right TopK list
+     * @return The merged list.
+     */
+    @Override
+    public NextKList add(NextKList left, NextKList right) {
+        if (!left.table.getSchema().equals(right.table.getSchema()))
+            throw new RuntimeException("The schemas do not match.");
+        int width = left.table.getSchema().getColumnCount();
+        List<IColumn> mergedCol = new ArrayList<IColumn>(width);
+        List<Integer> mergeOrder = this.recordOrder.getIntMergeOrder(left.table, right.table);
+        for (String colName : left.table.getSchema().getColumnNames())
+            mergedCol.add(this.mergeColumns(left.table.getColumn(colName),
```

Comment:
Please do this in two steps using a temporary variable for the result of mergeColumns.
This simplifies debugging if necessary.